### ISSUE TITLE
Error when reference module with specific symbols in module name

### ISSUE THREAD

**Issue by colderlord** [bridgedotnet/Bridge#3933]
I use UMD modules.
Error when reference module with specific symbols in module name

### Steps To Reproduce

https://deck.net/201f618e1b7dcea55381f100626fb8fb

```csharp
public class Program
{
    public static void Main()
    {
    }
}


[Module(ModuleType.UMD, "Class1, 1.0.0.0")]
public class Class1
{
    public Class2 Class2 {get;set;}
}
 
[Module(ModuleType.UMD, "Class2, 1.0.0.1")]
public class Class2
{
} 
```

Is it bug or it's just my fault, that i named like that?))

**Comment by geoffreymcgill** [comment]
If you can link us to some documentation that states `"Class1, 1.0.0.0"` is a valid JavaScript Module name, then we'll treat this as a bug.

**Comment by colderlord** [comment]
Module id is always "string", thats why i can write any symbols.
i have no problems when use `AMD`, `CommonJS`, but error only when i use `UMD` and in plaice that module defines in `global`....

in `Bridge` modules defines like that:
`root.Class1__1_0_0_0 = factory(root.Class2, 1.0.0.1);`
but if module will define like that: `root["Class1, 1.0.0.0"] = factory(root["Class2, 1.0.0.1"]);`
it will works.

as you can see in AMD module defines like that: `define("Class1, 1.0.0.0", ["Class2, 1.0.0.1"], factory);` Like string

**Comment by geoffreymcgill** [comment]
We'll have to check on `root.Class1__1_0_0_0 = factory(root.Class2, 1.0.0.1);`. That doesn't look correct and likely should be treated as a string. 

Thanks for point out this possible defect.

**Comment by geoffreymcgill** [comment]
A fix for this issue has been submitted. Once a review and final testing are complete, our plan is to include the fix in the incoming Bridge 17.8.0 release.

**Comment by fabriciomurta** [comment]
The fix looks good, it will now define UMD modules similar to the suggested format:
```
root["Class1, 1.0.0.0"] = factory(root["Class2, 1.0.0.1"]);
```
